Client/mods/deathmatch/logic/rpc/CElementRPCs.cpp

```diff
@@ -765,3 +766,15 @@ void CElementRPCs::SetElementOnFire(CClientEntity* pSource, NetBitStreamInterfac
 {
     pSource->SetOnFire(bitStream.ReadBit());
 }
+
+void CElementRPCs::SetElementCollidableWith(CClientEntity* pSource, NetBitStreamInterface& bitStream)
+{
+    ElementID ElementID;
+    bool      bCollidable;
+
+    bitStream.Read(ElementID);
+    bitStream.ReadBit(bCollidable);
```

Comment:
bitStream.Read can return false

Reply:
It cannot. The protocol guarantees that the message is fully received.

Reply:
> It cannot. The protocol guarantees that the message is fully received.

So what is the purpose of this function being a bool, and why does most of the code check if it returns true?

Reply:
Code == documentation
It has sense on the server side. The client reads server packets only.
Checks on client could be used to add a new data to a packet without increasing bitstream version, i think